test/default.nix

```diff
@@ -1,10 +1,10 @@
-let
-  haskellNix = (import ../default.nix {});
-in
-{ pkgs ? import nixpkgs nixpkgsArgs
+{ haskellNix ? import ../default.nix { defaultCompilerNixName = compiler-nix-name; inherit checkMaterialization; }
```

Comment:
I'd have thought removing the `defaultCompilerNixName` here should be safe? If we're successfully being explicit everywhere else.

Reply:
This allows us to run the tests for a compiler with:

```
cd test
nix-build -A cabal-simple --arg compiler-nix-name ghc865
```

Reply:
I meant in the arguments of `default.nix`, I buy the argument for leaving it as an argument of the overall function.

Reply:
compiler-nix-name is threaded through all the tests now without relying on defaultCompilerNixName